lib/trino-collect/src/main/java/io/trino/collect/cache/EvictableCache.java

```diff
@@ -125,18 +124,22 @@ public V get(K key)
     {
         Token<K> newToken = new Token<>(key);
         Token<K> token = tokens.computeIfAbsent(key, ignored -> newToken);
-        try {
-            return dataCache.get(token);
-        }
-        catch (Throwable e) {
-            if (newToken == token) {
+        if (token == newToken) {
```

Comment:
Could you split the PR in 2 commits?


One of the issues is that the `getAll(Iterable keys)` method does not provide back in the result map the same key objects as the ones provided in the method call.

This is being solved (I'm assuming) in `io.trino.collect.cache.EvictableCache#getAll` and tested in `TestEvictableLoadingCache`.

IMO the changes on the method `io.trino.collect.cache.EvictableCache#get` are not related to the first change.

Reply:
Both changes are closely related to each other: both `get` and `getAll` had same problem.
technically, i can split into two commits, but since the change is small overall, it wouldn't really help.